src/Components/TodoItem.tsx

```diff
@@ -0,0 +1,204 @@
+    setLoadingIds(currIds => [...currIds, todoId]);
+
+    updateTodo(todoId, { completed: !todo.completed })
+      .then(() => {
+        getTodos(USER_ID)
+          .then((todos) => {
+            setTodos(todos);
+          })
+          .catch(() => {
+            setErrorMessage(Notification.load);
+          })
+          .finally(() => {
+            setLoadingIds(currIds => currIds.filter(id => id !== todoId));
+          });
+      })
+      .catch(() => {
+        setErrorMessage(Notification.update);
+      });
+  };
+
+  const handleDoubleClick = () => {
+    setDoubleClicked(true);
+  };
+
+  const handleClickedFormSubmit = (
+    event: FormEvent,
+    title: string,
+    todoId: number,
+  ) => {
+    event.preventDefault();
+
+    if (title.trim() === '') {
+      setDoubleClicked(false);
+      setLoadingIds(currIds => [...currIds, todoId]);
+
+      deleteTodo(todoId)
+        .then(() => {
+          getTodos(USER_ID)
+            .then((todos) => {
+              setTodos(todos);
+            })
+            .catch(() => {
+              setErrorMessage(Notification.load);
+              setLoadingIds(currIds => currIds.filter(id => id !== todoId));
+            });
+        })
+        .catch(() => {
+          setErrorMessage(Notification.delete);
+          setLoadingIds(currIds => currIds.filter(id => id !== todoId));
+        });
+
+      return;
+    }
+
+    if (title !== todo.title) {
+      setLoadingIds(currIds => [...currIds, todoId]);
+
+      updateTodo(todoId, { title: editingTitle })
+        .then(() => {
+          getTodos(USER_ID)
+            .then((todos) => {
+              setTodos(todos);
+              setDoubleClicked(false);
+              setLoadingIds(currIds => currIds.filter(id => id !== todoId));
+            })
+            .catch(() => {
+              setErrorMessage(Notification.load);
+              setLoadingIds(currIds => currIds.filter(id => id !== todoId));
+            });
+        })
+        .catch(() => {
+          setErrorMessage(Notification.update);
+          setDoubleClicked(false);
+          setLoadingIds(currIds => currIds.filter(id => id !== todoId));
+        });
+
+      return;
+    }
+
+    setDoubleClicked(false);
+  };
+
+  const handleKeyUp = (key: string) => {
+    if (key === 'Escape') {
+      setDoubleClicked(false);
+      setEditingTitle(todo.title);
+    }
+  };
+
+  return (
+    <div
+      key={todo.id}
+      className={classNames('todo',
+        { completed: todo.completed })}
+    >
+      <label className="todo__status-label">
+        <input
+          type="checkbox"
+          className="todo__status"
+          checked={todo.completed}
+          onChange={() => {
+            handleToggle(todo.id);
+          }}
+        />
+      </label>
+
+      {doubleClicked ? (
+        <form
+          onSubmit={(event) => handleClickedFormSubmit(
+            event,
+            editingTitle,
+            todo.id,
+          )}
```

Comment:
make separate funcitons for all handlers and jsx and pass reference here
for example here something like that
check other similar cases

```
const handleClieckedFormSubmit = (id: string) => (event: React.FormEvent<HTMLFormElement>) => {
    // do all logic for submit
}
```

```suggestion
          onSubmit={handleClickedFormSubmit(todo.id)}
```